### PR TITLE
[Fix](executor)Release resource correctly when drop workload group

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes

Release resource correctly when drop workload group.

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

File: be/src/runtime/fragment_mgr.cpp [hunk 1]
```diff
@@ -685,20 +685,22 @@ Status FragmentMgr::_get_query_ctx(const Params& params, TUniqueId query_id, boo
...
-                query_ctx->set_query_scheduler(tg_id);
-
-                LOG(INFO) << "Query/load id: " << print_id(query_ctx->query_id())
-                          << ", use task group: " << task_group_ptr->debug_string()
-                          << ", is pipeline: " << ((int)is_pipeline)
-                          << ", enable cgroup soft limit: "
-                          << ((int)config::enable_cgroup_cpu_soft_limit);
+            taskgroup::TaskGroupPtr task_group_ptr = nullptr;
+            Status ret = tg_mgr->add_query_to_group(tg_id, query_ctx->query_id(), &task_group_ptr);
+            if (!ret.ok()) {
```
Comment (yiguolei) [inline be/src/runtime/fragment_mgr.cpp:690]: use  return_if_error

File: be/src/runtime/task_group/task_group_manager.cpp [hunk 2]
```diff
@@ -180,68 +180,72 @@ Status TaskGroupManager::upsert_cg_task_scheduler(taskgroup::TaskGroupInfo* tg_i
...
+    for (auto* ptr1 : task_sched_to_stop) {
         ptr1->stop();
     }
-    for (auto* ptr2 : scan_task_sche_to_del) {
+    for (auto* ptr2 : scan_task_sched_to_stop) {
         ptr2->stop();
     }
-    for (auto& ptr3 : non_pip_thread_pool_to_del) {
+    for (auto& ptr3 : non_pip_thread_pool_to_stop) {
         ptr3->shutdown();
```
Comment (yiguolei) [inline be/src/runtime/task_group/task_group_manager.cpp:226]: stop 和 shutdown 的区别是啥？

File: be/src/runtime/task_group/task_group.h [hunk 3]
```diff
@@ -141,6 +141,16 @@ class TaskGroup : public std::enable_shared_from_this<TaskGroup> {
         return _memory_limit > 0;
     }
 
+    void add_query(TUniqueId query_id) { _query_id_set.insert(query_id); }
```
Comment (yiguolei) [inline be/src/runtime/task_group/task_group.h:144]: Does these method have concurrent problem? If there are many queries to be added to the group?

File: be/src/runtime/task_group/task_group_manager.cpp [hunk 4]
```diff
@@ -180,68 +180,72 @@ Status TaskGroupManager::upsert_cg_task_scheduler(taskgroup::TaskGroupInfo* tg_i
...
+        ptr3->wait();
```
Comment (yiguolei) [inline be/src/runtime/task_group/task_group_manager.cpp:227]: why there is a wait here? and not wait at line 223? or line 220?

Comment (wangbo) [reply be/src/runtime/task_group/task_group_manager.cpp:227]: 这个是threadpool的标准用法，先标记poolshutdown，然后等没有活跃线程了，就都结束了。
wait就是等线程结束的过程。因为非pipeline目前没有封装scheduler，所以是手动这么处理的。
pipeline的scheduler调用stop的里边，也是做的同样的操作

File: be/src/runtime/task_group/task_group.h (hunk 3 above)
Comment (wangbo) [reply be/src/runtime/task_group/task_group.h:144]: 被taskgroupmgr里的_group_mutex锁住的

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buidlall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**